src/main/java/model/member/ExtendedBattleMember.java

```diff
@@ -432,18 +434,18 @@ private Weapon randomWeapon(Collection<String> types, Collection<Weapon> weaponP
         ArrayList<Weapon> weapons = new ArrayList<>();
         for (int i = getTier(); i > 0 && weapons.size() == 0; i--) {
             int k = i;
-            String rarity = Utility.getRandomRarity();
+            Rarity rarity = Utility.getRandomRarity();
             weapons = weaponPool.stream()
                     .filter(x -> x.getTier() == k)
-                    .filter(x -> x.getRarity().equals(rarity))
+                    .filter(x -> x.getRarity() == rarity)
                     .filter(this::checkRequirements)
                     .filter(x -> types.stream().anyMatch(y -> x.getSubTyp().equals(y)))
                     .collect(Collectors.toCollection(ArrayList::new));
 
             if (weapons.size() == 0) {
                 weapons = weaponPool.stream()
                         .filter(x -> x.getTier() == k)
-                        .filter(x -> x.getRarity().equals("gewöhnlich"))
+                        .filter(x -> x.getRarity() == Rarity.common)
```

Comment:
so what this code does is, that I try to get a weapon of random rarity which this battleMember can equip. If they can not equip any weapon of that rarity, I instead equip a common rarity item?

I think this is a flawed design. If the item is "too rare", it should try the next rarity until working its way to common